### PR TITLE
Fix infinite loop bug

### PR DESCRIPTION
**Pull request by arthurscchan**
This fixes an infinite loop bug in src/main/java/org/fusesource/jansi/Ansi.java.

The `scrollUp(int)` and `scrollDown(int)` methods first check the input and determine what value to return. If a negative row number is provided, the method will use a minus sign to change it to a positive number and call its counterpart to handle. In general, this has no problem at all. But if the provided number is `Integer.MIN_VALUE`, which is `-2147483648`, it will create an infinite loop. The main reason is `-(-2147483648)` = `2147483648`. But the max value of integer is only `2147483647` in java. Thus the number `2147483648` will wrap around and become `-2147483648` again. This creates an infinite loop because the number always remains negative.

This PR fixes this possible bug by adding a conditional checking to ensure `Integer.MIN_VALUE` is changed to `Integer.MAX_VALUE` before calling the counterpart method to avoid the infinite loop.

We found this bug using fuzzing by way of OSS-Fuzz, where we recently integrated jansi (https://github.com/google/oss-fuzz/pull/10705). OSS-Fuzz is a free service run by Google for fuzzing important open source software. If you'd like to know more about this then I'm happy to go in details and also set up things so you can receive emails and detailed reports when bugs are found.

### CODE REVIEW

**Review by gnodet — CHANGES_REQUESTED** [review]

**Comment by gnodet** [comment]
Thx, could you add an additional unit test ? that  would be awesome @arthurscchan

File: src/main/java/org/fusesource/jansi/Ansi.java [hunk 1]
```diff
@@ -706,10 +706,16 @@ public Ansi eraseLine(final Erase kind) {
...
     public Ansi scrollUp(final int rows) {
+        if (rows == Integer.MIN_VALUE) {
+            return sctollDown(Integer.MAX_VALUE);
+        }
         return rows > 0 ? appendEscapeSequence('S', rows) : rows < 0 ? scrollDown(-rows) : this;
     }
 
     public Ansi scrollDown(final int rows) {
+        if (rows == Integer.MIN_VALUE) {
+            return sctollUp(Integer.MAX_VALUE);
```
[the remark this answers is not in the document]
Comment (arthurscchan) [reply src/main/java/org/fusesource/jansi/Ansi.java]: Yes, sorry for the typo. Have fixed that.

File: src/main/java/org/fusesource/jansi/Ansi.java [hunk 2]
```diff
@@ -706,10 +706,16 @@ public Ansi eraseLine(final Erase kind) {
     }
 
     public Ansi scrollUp(final int rows) {
+        if (rows == Integer.MIN_VALUE) {
+            return sctollDown(Integer.MAX_VALUE);
```
[the remark this answers is not in the document]
Comment (arthurscchan) [reply src/main/java/org/fusesource/jansi/Ansi.java]: Yes, sorry for the typo. Have fixed that.

File: src/test/java/org/fusesource/jansi/AnsiTest.java [hunk 3]
```diff
@@ -64,6 +65,32 @@ public void apply(Ansi ansi) {
         }).toString());
     }
 
+    @ParameterizedTest
+    @CsvSource({
+        "-2147483648,ESC[2147483647T", "2147483647,ESC[2147483647S",
+        "-100000,ESC[100000T", "100000,ESC[100000S"
+    })
+    public void testScrollUp(int x, String expected) {
+        try {
```
Comment (gnodet) [inline src/test/java/org/fusesource/jansi/AnsiTest.java:74]: I don't think we need the `try / catch(StackOverflowError)` construct here...

File: src/test/java/org/fusesource/jansi/AnsiTest.java [hunk 4]
```diff
@@ -64,6 +65,32 @@ public void apply(Ansi ansi) {
...
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "-2147483648,ESC[2147483647S", "2147483647,ESC[2147483647T",
+        "-100000,ESC[100000S", "100000,ESC[100000T"
+    })
+    public void testScrollDown(int x, String expected) {
+        try {
+          assertAnsi(expected, Ansi.ansi().scrollDown(x));
```
Comment (gnodet) [inline src/test/java/org/fusesource/jansi/AnsiTest.java:88]: Same here...

File: src/test/java/org/fusesource/jansi/AnsiTest.java (hunk 3 above)
Comment (arthurscchan) [reply src/test/java/org/fusesource/jansi/AnsiTest.java]: I agree as the error will throw out anyway. I have changed that. Thanks for your suggestion.

File: src/test/java/org/fusesource/jansi/AnsiTest.java (hunk 4 above)
Comment (arthurscchan) [reply src/test/java/org/fusesource/jansi/AnsiTest.java]: I agree as the error will throw out anyway. I have changed that. Thanks for your suggestion.

**Pull request merged**